Sources/main.cpp

```diff
@@ -19,6 +22,33 @@
 #include "core/FastIsType.h"
 #include "AutogenMetadata.h"
 
+QFile logs("error.log");
+
+void myMessageOutput(QtMsgType type, const QMessageLogContext &context, const QString &msg)
+{
+    QTextStream logstream(&logs);
+    QString datetime = QDateTime::currentDateTime().toString();
+    switch (type) {
+    case QtDebugMsg:
+        logstream << "Debug, " << datetime << ": " << msg << " " << context.file << " " << context.line << " " << context.function << endl;
+        break;
+    case QtWarningMsg:
+        logstream << "Warning, " << datetime << ": " << msg << " " << context.file << " " << context.line << " " << context.function << endl;
```

Comment:
1. `datetime << ": " << msg << " " << context.file << " " << context.line << " " << context.function << endl;` is copy-pase
2. File, line and function should be before message, because it will be inconvenient to read log otherwise.